src/transformers/modeling_utils.py

```diff
@@ -1718,6 +1718,42 @@ def save_pretrained(
             for ignore_key in self._keys_to_ignore_on_save:
                 if ignore_key in state_dict.keys():
                     del state_dict[ignore_key]
+        if safe_serialization:
+            # Safetensors does not allow tensor aliasing.
+            # We're going to remove aliases before saving
+            ptrs = collections.defaultdict(list)
+            for name, tensor in state_dict.items():
+                ptrs[tensor.data_ptr()].append(name)
+
+            # These are all the pointers of shared tensors.
+            shared_ptrs = {ptr: names for ptr, names in ptrs.items() if len(names) > 1}
+            warn_names = set()
+            for _, names in shared_ptrs.items():
+                # Removing the keys which are declared as known duplicates on
+                # load. This allows to make sure the name which is kept is consistent.
+                if self._keys_to_ignore_on_load_missing is not None:
+                    for name in names:
+                        for pat in self._keys_to_ignore_on_load_missing:
+                            if re.search(pat, name):
+                                if name in state_dict:
+                                    del state_dict[name]
+
+                # When not all duplicates have been cleaned
+                # Still remove those keys, but put a clear warning
+                # Since if the link between tensors was done at runtime
+                # then `from_pretrained` will still not get the key back
+                # Leading to random tensor. With a proper warning.
+                found = 0
+                for name in names:
+                    if name in state_dict:
+                        found += 1
+                        if found > 1:
+                            del state_dict[name]
+                            warn_names.add(name)
+            if warn_names:
+                logger.warning(
```

Comment:
```suggestion
                logger.warn_once(
```
To avoid the warning popping up 500 times when someone is training their model and regularly checkpointing.